src/mono/browser/runtime/marshal-to-js.ts

```diff
@@ -276,8 +276,9 @@ export function end_marshal_task_to_js (args: JSMarshalerArguments, res_converte
     // get the synchronous result
     const promise = try_marshal_sync_task_to_js(res, type, res_converter);
 
-    // make sure we got the result
-    mono_assert(promise !== false, () => `Expected synchronous result, got: ${type}`);
+    if (promise === false) {
```

Comment:
This change probably wrong, the assert was there for a reason (which I only vaguely remember now). 
If this is something we could revert, it would be great. 
If not, I would like to ask what scenario prompted this change, so that I can think it thru. Thank you.